doc/mclag/MCLAG_Enhancements_HLD.md

```diff
@@ -0,0 +1,810 @@
+
+![Port Isolation group processing](images/PortIsolationAdd.png "Port Isolation group processing")
+
+**Figure 6: Port Isolation group processing**
+
+# 5 Serviceability and Debug
+- Existing mclagdctl dump commands output will be extended with additional debug state information.
+
+Mclagdctl dump state command is added with new field to show the sync state.
+```
+Example output:
+root@sonic:/home/admin# mclagdctl -i 5 dump state
+The MCLAG's keepalive is: OK
+**MCLAG info sync is: completed**
+Domain id: 5
+Local Ip: 192.168.1.1
+Peer Ip: 192.168.1.2
+Peer Link Interface: PortChannel30
+Keepalive time: 1
+sesssion Timeout : 15
+Peer Link Mac: b8:6a:97:73:6c:96
+Role: Active
+MCLAG Interface: PortChannel50
+root@sonic:/home/admin#
+```
+  Mclagdctl is extended with new option to dump the debug counters information.
+
+```
+Example Output:
+root@sonic:/home/admin# mclagdctl -i 5 dump debug counters
+ICCP session down:  1
+Peer link down:     0
+Warmboot:           0
+
+ICCP to MclagSyncd      TX_OK       TX_ERROR
+------------------      -----       --------
+PortIsolation           13          0
+MacLearnMode            1           0
+FlushFdb                1           0
+SetIfMac                0           0
+SetFdb                  226003      0
+SetL2mc                 0           0
+TrafficDistEnable       1           0
+TrafficDistDisable      1           0
+SetIccpState            2           0
+SetIccpRole             1           0
+SetSystemId             0           0
+DelIccpInfo             0           0
+SetRemoteIntfSts        9           0
+DelRemoteIntf           0           0
+
+MclagSyncd to          ICCP  RX_OK       RX_ERROR
+------------------     -----------       --------
+FdbChange               161616           0
+CfgMclag                1                0
+CfgMclagIface           1                0
+
+ICCP to Peer   TX_OK        RX_OK        TX_ERROR    RX_ERROR
+------------   -----        -----        --------    --------
+SysConfig      2            2            0           0
+AggrConfig     2            1            0           0
+AggrState      7            9            0           0 
+MacInfo        160896       142183       0           0
+ArpInfo        0            0            0           0
+L2mcInfo       0            0            0           0
+PoInfo         2            1            0           0
+PeerLinkInfo   2            2            0           0
+Heartbeat      2533         2533         0           0
+Nak            0            0            0           0
+SyncData       2            2            0           0
+SyncReq        1            0            0           0
+Warmboot       0            0            0           0
+IfUpAck        4            3            0           0
+
+```
+
+- MCLAG debug guide can be referred for trouble shooting issues. Below is the link to debug guide.
+
+  https://docs.google.com/document/d/1exNQ1po7TYmVtctq59aSUeBsZYKctbHMkwStNVNgSrU/edit 
+
+# 6 Warm Boot Support
```

Comment:
When a peer is going through warmboot, how do we ensure that ICCP keepalives dont time out?